### PR TITLE
[typescript][angular]: Fix for issue #7711

### PR DESCRIPTION
**Pull request by liuy97**
Usehttpclient should work with interface. It also fixes tslint.

### PR checklist

- [x] Read the [contribution guidelines](https://github.com/swagger-api/swagger-codegen/blob/master/CONTRIBUTING.md).
- [x] Ran the shell script under `./bin/` to update Petstore sample so that CIs can verify the change. (For instance, only need to run `./bin/{LANG}-petstore.sh` and `./bin/security/{LANG}-petstore.sh` if updating the {LANG} (e.g. php, ruby, python, etc) code generator or {LANG} client's mustache templates). Windows batch files can be found in `.\bin\windows\`.
- [ ] Filed the PR against the correct branch: `3.0.0` branch for changes related to OpenAPI spec 3.0. Default: `master`.
- [x] Copied the [technical committee](https://github.com/swagger-api/swagger-codegen/#swagger-codegen-technical-committee) to review the pull request if your PR is targeting a particular programming language.

### Description of the PR

(details of the change, additional tests that have been done, reference to the issue for tracking, etc)

@TiFu (2017/07) @taxpon (2017/07) @sebastianhaas (2017/07) @kenisteward (2017/07) @Vrolijkx (2017/09) @macjohnny (2018/01)

### CODE REVIEW

File: modules/swagger-codegen/src/main/resources/typescript-angular/apiInterface.mustache [hunk 1]
```diff
@@ -14,9 +23,9 @@ import { Configuration }                                     from '../configurat
     */
 {{/description}}
 export interface {{classname}}Interface {
-    defaultHeaders: Headers;
+    defaultHeaders: {{#useHttpClient}}Http{{/useHttpClient}}Headers;
     configuration: Configuration;
-    [others: string]: any;
+    {{^useHttpClient}}[others: string]: any;{{/useHttpClient}}
```
Comment (macjohnny) [inline modules/swagger-codegen/src/main/resources/typescript-angular/apiInterface.mustache]: can anyone explain why this is necessary / not necessary anymore?

Comment (wing328) [reply modules/swagger-codegen/src/main/resources/typescript-angular/apiInterface.mustache]: @macjohnny I'm not sure either. The `useHttpClient` tag is introduced for Typescript version 4.3 or later: https://github.com/swagger-api/swagger-codegen/blob/157e6b7fab4c4b3cddee88fb1100271c2365a6da/modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/TypeScriptAngularClientCodegen.java#L127

Looks like that line is needed for Angular 2.x but not Angular 4.3 or later?

**Comment by macjohnny** [comment]
I think this PR should be closed in favor of https://github.com/swagger-api/swagger-codegen/pull/7975
do you all agree?

**Comment by kenisteward** [comment]
Yes agreed

**Comment by liuy97** [comment]
Yes

**Pull request closed without merging**